types/sqs-consumer/index.d.ts

```diff
@@ -1,29 +1,33 @@
-// Type definitions for sqs-consumer 3.5
+// Type definitions for sqs-consumer 3.7
 // Project: https://github.com/BBC/sqs-consumer
-// Definitions by: Daniel Chao <http://dchao.co/>, Eric Byers <http://www.github.com/EricByers>
+// Definitions by: Daniel Chao <http://dchao.co/>
+//                 Eric Byers <http://github.com/EricByers>
+//                 Ezinwa Okpoechi <http://github.com/BrainMaestro>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /// <reference types="node" />
 import { SQS } from "aws-sdk";
 
-export type ConsumerDone = (error?: Error) => void;
+export type ConsumerDone = (error?: SQSError) => void;
 
 export interface Options {
-  queueUrl: string;
-  handleMessage(message: SQS.Message, done: ConsumerDone): any;
-  region?: string;
-  attributeNames?: string[];
-  messageAttributeNames?: string[];
-  batchSize?: number;
-  visibilityTimeout?: number;
-  waitTimeSeconds?: number;
-  authenticationErrorTimeout?: number;
-  sqs?: SQS;
+    queueUrl: string;
+    handleMessage(message: SQS.Message, done: ConsumerDone): any;
+    region?: string;
+    attributeNames?: string[];
+    messageAttributeNames?: string[];
+    batchSize?: number;
+    visibilityTimeout?: number;
+    waitTimeSeconds?: number;
+    authenticationErrorTimeout?: number;
+    sqs?: SQS;
 }
 
-export interface Consumer extends NodeJS.EventEmitter {
-  start(): void;
-  stop(): void;
-}
+export class SQSError extends Error {}
```

Comment:
`SQSError` isn't an exported value in sqs-consumer.

Reply:
It's not, but like I mentioned before, you can still get an `SQSError` from certain functions, and exporting it will give a way of differentiating the error from a regular one. I think it should even be exported in sqs-consumer

Reply:
`SQSError` isn't a *value* that's exported from this module, meaning that you have no way of importing this value. The current typing is basically asserting that you can write this equivalent in JavaScript:

```js
var SQSError = require('sqs-consumer').SQSError
```

The object that `require('sqs-consumer')` returns doesn't have an `SQSError` property, so in this case `SQSError` would be `undefined`.

You'd need to refactor this to be an `interface` rather than a `class`